src/main/java/org/truffleruby/core/hash/BucketsStrategy.java

```diff
@@ -194,14 +194,17 @@ public static void copyInto(RubyContext context, RubyHash from, RubyHash to) {
         assert HashOperations.verifyStore(context, to);
     }
 
-
     public static void removeFromSequenceChain(RubyHash hash, Entry entry) {
         if (entry.getPreviousInSequence() == null) {
             assert hash.firstInSequence == entry;
             hash.firstInSequence = entry.getNextInSequence();
         } else {
             assert hash.firstInSequence != entry;
             entry.getPreviousInSequence().setNextInSequence(entry.getNextInSequence());
+            final Entry nextInSequence = entry.getNextInSequence();
+            if (nextInSequence != null) {
+                nextInSequence.setPreviousInSequence(entry.getPreviousInSequence());
+            }
```

Comment:
There is missing logic here to update `hash.lastInSequence`, I'll fix it. With that then it's perfectly symmetrical.